openfe/protocols/openmm_septop/base.py

```diff
@@ -0,0 +1,1436 @@
+            simulation_settings=settings['equil_simulation_settings'],
+            output_settings=unfrozen_outsettings,
+            temperature=settings['thermo_settings'].temperature,
+            barostat_frequency=settings[
+                'integrator_settings'].barostat_frequency,
+            timestep=settings['integrator_settings'].timestep,
+            equil_steps_nvt=equil_steps_nvt,
+            equil_steps_npt=equil_steps_npt,
+            prod_steps=prod_steps_npt,
+            verbose=self.verbose,
+            shared_basepath=self.shared_basepath,
+        )
+        state = simulation.context.getState(getPositions=True
+                                            ,enforcePeriodicBox=True)
+        equilibrated_positions = state.getPositions(asNumpy=True)
+        box = state.getPeriodicBoxVectors()
+
+        # cautiously delete out contexts & integrator
+        del simulation.context, integrator
+
+        return equilibrated_positions, box
+
+    def _get_states(
+            self,
+            alchemical_system: openmm.System,
+            positions: openmm.unit.Quantity,
+            box_vectors: Optional[openmm.unit.Quantity],
+            settings: dict[str, SettingsBaseModel],
+            lambdas: dict[str, npt.NDArray],
+            solvent_comp: Optional[SolventComponent],
+    ) -> tuple[list[SamplerState], list[ThermodynamicState]]:
+        """
+        Get a list of sampler and thermodynmic states from an
+        input alchemical system.
+
+        Parameters
+        ----------
+        alchemical_system : openmm.System
+          Alchemical system to get states for.
+        positions : openmm.unit.Quantity
+          Positions of the alchemical system.
+        box_vectors : Optional[openmm.unit.Quantity]
+          Box vectors of the alchemical system.
+        settings : dict[str, SettingsBaseModel]
+          A dictionary of settings for the protocol unit.
+        lambdas : dict[str, npt.NDArray]
+          A dictionary of lambda scales.
+        solvent_comp : Optional[SolventComponent]
+          The solvent component of the system, if there is one.
+
+        Returns
+        -------
+        sampler_states : list[SamplerState]
+          A list of SamplerStates for each replica in the system.
+        cmp_states : list[ThermodynamicState]
+          A list of ThermodynamicState for each replica in the system.
+        """
+        alchemical_state = SepTopParameterState.from_system(alchemical_system)
+
+        # Set up the system constants
+        temperature = settings['thermo_settings'].temperature
+        pressure = settings['thermo_settings'].pressure
+        constants = dict()
+        constants['temperature'] = ensure_quantity(temperature, 'openmm')
+        if solvent_comp is not None:
+            constants['pressure'] = ensure_quantity(pressure, 'openmm')
+
+        # Get the composable states
+        composable_states = [
+            alchemical_state
+        ]
```

Comment:
Nitpick:
I would just move this into the call like here: https://github.com/OpenFreeEnergy/openfe/blob/main/openfe/protocols/openmm_afe/base.py#L678

That way it's easier to diff the two and merge things later.